kvserver: alias range_id in flow control integration tests

kvoli opened the pull request:
Earlier commits are mechanical, reformatting and deduplicating SQL queries
used by the flow control integration tests.

---

Previously, the `range_id` of scratch ranges used in various
`TestFlowControl.*` tests would be printed into the testdata file. This
made it tedious to introduce new system tables, which in turn may create
new ranges and necessitate regenerating the
`test_flow_control_integration` testdata files to pick up the higher
default scratch range id.

Alias the range id to letters in `a,b,...,z` so that regardless of the
underlying scratch range id changing, the aliased scratch range id
remains unchanged. For example:

An old example:

```sql
SELECT
  range_id, count(*) AS streams
FROM
  crdb_internal.kv_flow_control_handles
GROUP BY
  (range_id)
ORDER BY
  streams DESC;

  range_id | stream_count
-----------+---------------
  75       | 3
  76       | 3
```

Now looks like:

```sql
SELECT
  chr(96 + DENSE_RANK() OVER (ORDER BY range_id)) as range_id,
  count(*) AS streams
FROM
  crdb_internal.kv_flow_control_handles
GROUP BY
  range_id
ORDER BY
  range_id;

  range_id | stream_count
-----------+---------------
  a        | 3
  b        | 3
```

Resolves: https://github.com/cockroachdb/cockroach/issues/137727
Release note: None

kvoli committed:
kvserver: re-use flow control test query strings

These were duplicated around many tests in
`flow_control_integration_test.go`. De-duplicate.

Part of: #137727
Release note: None

kvoli committed:
kvserver: refmt flow control integration test sql queries

The SQL query strings which are reused across several
`TestFlowControl.*` tests were inconsistently formatted. Format them
consistently, although not following any style guide.

Part of: #137727
Release note: None

kvoli committed:
kvserver: alias range_id in flow control integration tests

Previously, the `range_id` of scratch ranges used in various
`TestFlowControl.*` tests would be printed into the testdata file. This
made it tedious to introduce new system tables, which in turn may create
new ranges and necessitate regenerating the
`test_flow_control_integration` testdata files to pick up the higher
default scratch range id.

Alias the range id to letters in `a,b,...,z` so that regardless of the
underlying scratch range id changing, the aliased scratch range id
remains unchanged. For example:

An old example:

```sql
SELECT
  range_id, count(*) AS streams
FROM
  crdb_internal.kv_flow_control_handles
GROUP BY
  (range_id)
ORDER BY
  streams DESC;

  range_id | stream_count
-----------+---------------
  75       | 3
  76       | 3
```

Now looks like:

```sql
SELECT
  chr(96 + dense_rank() OVER (ORDER BY range_id)) as range_id,
  count(*) AS streams
FROM
  crdb_internal.kv_flow_control_handles
GROUP BY
  range_id
ORDER BY
  range_id;

  range_id | stream_count
-----------+---------------
  a        | 3
  b        | 3
```

Resolves: #137727
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/138085)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 7 of 7 files at r1, 77 of 78 files at r2, 60 of 60 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/138085#-:-OG2aItW808ziQki9OK0:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli)



<!-- Sent from Reviewable.io -->

kvoli commented:
`TestFlowControlAdmissionPostSplitMerge` is failing on extendedCI -- investigating.

kvoli commented:
Rebased to resolve merge conflicts.

kvoli commented:
I'm going to wipe the v1 rac tests and rebase on-top, the failure is related but not something we need to test, given we won't be running the v1 code post 24.3.

kvoli committed:
kvserver: delete v1 flow control integration tests

These tests are now redundant, as users will run only the RACv2 protocol
post-upgrade to 24.3.

Part of: #136529
Release note: None

kvoli commented:
Pushed a commit to enable verbose logging.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 17 of 17 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/138085#-:-OGqBY6j5E4blCTkSOvt:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli)



<!-- Sent from Reviewable.io -->

pav-kv commented on pkg/kv/kvserver/flow_control_integration_test.go:
```diff
@@ -196,6 +196,15 @@ func TestFlowControlRangeSplitMergeV2(t *testing.T) {
 			defer h.close(makeV2EnabledTestFileName(v2EnabledWhenLeaderLevel, mode, "split_merge"))
 
+			h.comment(`-- Enable verbose logging for relevant RACv2 code files.`)
+			h.query(n1, `
```
Do we need to print this to output? Would something like [this](https://github.com/cockroachdb/cockroach/blob/56faadf8ddb58c21678c2b4f8df5418931656887/pkg/ccl/changefeedccl/changefeed_test.go#L6756-L6757) do the trick?

pav-kv replied:
Alternative: #139484.

pav-kv committed:
kvserver: add vlogging to flaky tests

Epic: none
Release note: none

kvoli committed:
kvserver: print all flow control integration test puts

Previously, some `TestFlowControl.*` put requests were not printed to
the datadriven test file, only printed to the test log. This commit
updates these tests:

```
TestFlowControlRangeSplitMergeV2
TestFlowControlAdmissionPostSplitMergeV2
```

To also print to the datadriven test file on puts. The test log is also
still written to.

Epic: None
Release note: None

kvoli committed:
kvserver: issue min size writes in test flow control tests

The following tests are updated to issue 1 byte sized puts, which are
overridden via a testing knob to be 1 MiB. Some of these tests have
previously flaked due to disconnected streams, hypothesized to be a
result of CPU starvation in the test.

```
TestFlowControlBasicV2
TestFlowControlAdmissionPostSplitMergeV2
TestFlowControlRangeSplitMergeV2
TestFlowControlBlockedAdmissionV2
TestFlowControlCrashedNodeV2
TestFlowControlRaftMembershipV2
TestFlowControlRaftMembershipRemoveSelfV2
TestFlowControlUnquiescedRangeV2
TestFlowControlClassPrioritizationV2
TestFlowControlTransferLeaseV2
TestFlowControlLeaderNotLeaseholderV2
TestFlowControlGranterAdmitOneByOneV2
```

Note, some of these tests did not have any recent failures associated
with them and the load reduction is preventative.

Informs: #137510
Informs: #138103

Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/138085#-:-OH9OKXt3S35FLfv3jHk:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli and @pav-kv)



<!-- Sent from Reviewable.io -->

kvoli commented:
bors r=sumeerbhola

kvoli commented:
blathers backport 25.1
